gen/go/eventschema/getters.go

```diff
@@ -71,87 +71,132 @@ func QueryableEventList() (string, error) {
 	return sb.String(), nil
 }
 
-// TableSchema returns a CSV description of the event table schema.
-// This explains to the query writer which field are available.
-// This must not be confused with ViewSchema which returns the Athena SQL
-// statements used to build a view extracting content from raw event_data.
-func (event *Event) TableSchema() (string, error) {
+// TableSchemaDetails describe Athena view schema.
+type TableSchemaDetails struct {
+	// Name is view name
+	Name string
+	// SQLViewName is SQL compatible table name.
+	SQLViewName string
+	// Description is Athena view description.
+	Description string
+	// Columns contains information about columns schema.
+	Columns []*ColumnSchemaDetails
+}
+
+func (d *TableSchemaDetails) CreateView() string {
 	sb := strings.Builder{}
-	sb.WriteString("column_name, column_type, description\n")
-	sb.WriteString("event_date, date, is the event date\n")
-	sb.WriteString("event_time, timestamp, is the event time\n")
-	for _, prop := range event.Fields {
-		line, err := prop.TableSchema([]string{prop.Name})
-		if err != nil {
-			return "", trace.Wrap(err)
-		}
-		sb.WriteString(line)
+	sb.WriteString("SELECT\n")
+	sb.WriteString("  event_date, event_time\n")
+	for _, v := range d.Columns {
+		sb.WriteString(viewSchemaLine(v.NameJSON(), v.NameSQL(), v.Type))
 	}
```

Comment:
Is this safe w/r/t SQL injections? Can any part of this be user-provider? I would add a comment explaining why using string builder to assemble a query is safe here.

Same for all other places where string builder is used.

Reply:
This was exampled in https://github.com/gravitational/teleport/pull/33459/files/9f11cf6028aa27ce77be6381d2728882cc7f84c6#diff-088c806be4a3b1cd50cb8c07a9cffdda170738e3f746a12863f27aa86d32586b)

Namely Access Monitoring feature allows to run any arbitrary SQL query by a user where the Table permissions are handled by IAM.